### ISSUE TITLE
Inlining in a layout file?

### ISSUE THREAD

**Issue by egermani** [miloshadzic/middleman-inline#4]
Things seem to work okay if I call `inline_css` in a template, but when I include that method in a layout and try to build, I get a `stack level too deep`. If I had to guess I'd say this is because the inlining function isn't complete by the time the layout template pulls in the child template -- does that make sense?

Full error message:

```ruby
/Users/admin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/contracts-0.13.0/lib/contracts/validators.rb:104:in `make_validator': stack level too deep (SystemStackError)
	from /Users/admin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/contracts-0.13.0/lib/contracts.rb:189:in `valid?'
	from /Users/admin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/contracts-0.13.0/lib/contracts/builtin_contracts.rb:104:in `block in valid?'
	from /Users/admin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/contracts-0.13.0/lib/contracts/builtin_contracts.rb:103:in `any?'
	from /Users/admin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/contracts-0.13.0/lib/contracts/builtin_contracts.rb:103:in `valid?'
	from /Users/admin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/contracts-0.13.0/lib/contracts/validators.rb:53:in `block (2 levels) in <module:Validators>'
	from /Users/admin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/contracts-0.13.0/lib/contracts/call_with.rb:18:in `[]'
	from /Users/admin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/contracts-0.13.0/lib/contracts/call_with.rb:18:in `block in call_with'
	from /Users/admin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/contracts-0.13.0/lib/contracts/call_with.rb:13:in `times'
	 ... 7540 levels...
```

**Comment by miloshadzic** [comment]
Hey, I'm a bit busy these days but I've written a quick test to see if things blow up. It seems to be working ok (I also use this in a layout on my site). Could it be that it's something specific to your gem/site configuration? I don't use contracts for anything for example and that's where things look like they're blowing up in you case.

If you put up some parts of your configuration and/or layouts maybe I could help out.

**Comment by egermani** [comment]
Hi Milos,

Thanks for your reply. Your note about not using contracts was helpful -- I'm using middleman 4.2.1, which has contracts as a dependency. So perhaps your extension isn't fully compatible with middleman's latest. I should be able to come up with a workaround that'll work for my purposes.

**Comment by miloshadzic** [comment]
Hm, I doubt that's the case –– I just updated the version on my blog to 4.2.1(it used to be 4.2.0) and It works.

**Comment by egermani** [comment]
Oh, that is interesting. So your doubt gave me confidence that I'd messed something up in my config, and I started stripped stuff away.

In the end, the issue had nothing to do with your inliner. I had a custom layout that was to be applied only during the build process, and instead of applying it to just the subfolder that needed it, I gave it a path of `*`. This seems to have put everything in the source folder through the layout rendering process, including the stylesheets. Now that I've targeted that layout correctly, everything's happy, and things are being inlined no trouble.

Thanks!
